alacritty_terminal/src/term/mod.rs

```diff
@@ -320,9 +498,65 @@ impl<T> Term<T> {
             selection: None,
             cell_width: size.cell_width as usize,
             cell_height: size.cell_height as usize,
+            is_fully_damaged: true,
+            damage,
         }
     }
 
+    #[must_use]
+    pub fn damage(&mut self, selection: Option<SelectionRange>) -> TermDamage<'_> {
+        // Ensure that we damage entire terminal in insert mode.
+        if self.mode.contains(TermMode::INSERT) {
+            self.is_fully_damaged = true;
+        }
```

Comment:
This doesn't make any sense to me, why would we check `INSERT` when queried for damage?

Applications might enable/disable `INSERT` multiple times per frame, so no matter what the current state of the flag is we can't use it to make any decision on what has happened since the last frame.

Maybe you want to add this code in the mode switch logic?

Reply:
There's a code in switch logic already. I just want to always damage if mode `INSERT` is present, mode change only helps with one frame, however if you won't toggle it on the next frame it won't help.